test/integration/kubernetes_deploy_test.rb

```diff
@@ -1287,6 +1287,129 @@ def test_missing_name_on_unknown_resource_prints_metadata_but_not_body
+    end
+    assert_deploy_success(result)
+
+    assert_logs_match_all([
+      "Successfully deployed 2 resource",
+      "Successful resources",
+      %r{PersistentVolumeClaim/with-storage-class\s+Pending},
+      %r{PersistentVolumeClaim/without-storage-class\s+Bound},
+    ], in_order: true)
+
+  ensure
+    kubeclient.delete_persistent_volume(pvname)
+    storage_v1_kubeclient.delete_storage_class(storage_class_name)
+  end
+
+  def test_pvc_immediate_bind
+    pvname = "local0003"
+    storage_class_name = "k8s-deploy-test-immediate-bind"
+
+    result = deploy_fixtures("pvc", subset: ["wait_for_first_consumer_storage_class.yml"]) do |fixtures|
+      sc = fixtures["wait_for_first_consumer_storage_class.yml"]["StorageClass"].first
+      sc["metadata"]["name"] = storage_class_name
+      sc["volumeBindingMode"] = "Immediate"
+    end
+    assert_deploy_success(result)
+
+    TestProvisioner.prepare_pv(pvname, storage_class_name: storage_class_name)
+    result = deploy_fixtures("pvc", subset: ["pvc.yml"]) do |fixtures|
+      pvc = fixtures["pvc.yml"]["PersistentVolumeClaim"].first
+      pvc["spec"]["storageClassName"] = storage_class_name
+    end
+    assert_deploy_success(result)
+
+    assert_logs_match_all([
+      "Successfully deployed 2 resource",
+      "Successful resources",
+      %r{PersistentVolumeClaim/with-storage-class\s+Bound},
+      %r{PersistentVolumeClaim/without-storage-class\s+Bound},
+    ], in_order: true)
+
+  ensure
+    kubeclient.delete_persistent_volume(pvname)
+    storage_v1_kubeclient.delete_storage_class(storage_class_name)
+  end
+
+  def test_pvc_no_pv
+    storage_class_name = "k8s-deploy-test-no-pv"
+
+    result = deploy_fixtures("pvc", subset: ["wait_for_first_consumer_storage_class.yml"]) do |fixtures|
+      sc = fixtures["wait_for_first_consumer_storage_class.yml"]["StorageClass"].first
+      sc["metadata"]["name"] = storage_class_name
+    end
+    assert_deploy_success(result)
+
+    result = deploy_fixtures("pvc", subset: ["pvc.yml", "pod.yml"]) do |fixtures|
+      pvc = fixtures["pvc.yml"]["PersistentVolumeClaim"].first
+      pvc["spec"]["storageClassName"] = storage_class_name
+    end
+    assert_deploy_failure(result)
+
+    assert_logs_match_all([
+      "Failed to deploy 1 priority resource",
+      "Pod/pvc: TIMED OUT (timeout: 20s)",
+    ], in_order: true)
+
+  ensure
+    storage_v1_kubeclient.delete_storage_class(storage_class_name)
+  end
+
+  def test_pvc_no_pv_or_sc
+    storage_class_name = "k8s-deploy-test-no-pv-or-sc"
+
+    result = deploy_fixtures("pvc", subset: ["pvc.yml", "pod.yml"]) do |fixtures|
+      pvc = fixtures["pvc.yml"]["PersistentVolumeClaim"].first
+      pvc["spec"]["storageClassName"] = storage_class_name
+    end
+    assert_deploy_failure(result)
+
+    assert_logs_match_all([
+      "Failed to deploy 1 priority resource",
+      "PersistentVolumeClaim/with-storage-class: TIMED OUT (timeout: 20s)",
```

Comment:
Is this test taking a full 20s? Can we use a custom timeout to cut that down?

Reply:
Good eye, it does :(. Should be able to decrease it, just haven't tried yet.

Reply:
This is already using a custom timeout via a `kubernetes-deploy.shopify.io/timeout-override`:
https://github.com/Shopify/kubernetes-deploy/blob/ef17097b42b5875e15c138b74f7a9eb80485af9c/test/fixtures/pvc/pvc.yml#L5

The default was 300s - so I shortened it to 20s. I chose 20s to be sure it wasn't so short as to create a false positive - but 20s is still pretty generous.